sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/ext/RawTransactionUnsafeExt.kt

```diff
@@ -0,0 +1,16 @@
+package cash.z.ecc.android.sdk.internal.model.ext
+
+import cash.z.ecc.android.sdk.internal.model.TransactionStatus
+import cash.z.ecc.android.sdk.model.ZcashNetwork
+import co.electriccoin.lightwallet.client.model.RawTransactionUnsafe
+import co.electriccoin.lightwallet.client.model.RawTransactionUnsafe.MainChain
+import co.electriccoin.lightwallet.client.model.RawTransactionUnsafe.Mempool
+import co.electriccoin.lightwallet.client.model.RawTransactionUnsafe.OrphanedBlock
+
+internal fun RawTransactionUnsafe.toTransactionStatus(network: ZcashNetwork): TransactionStatus {
+    return when (this) {
+        is MainChain -> TransactionStatus.Mined(height.toBlockHeight(network))
+        is Mempool -> TransactionStatus.NotInMainChain
+        is OrphanedBlock -> TransactionStatus.TxidNotRecognized
```

Comment:
This should be `NotInMainChain` - we got the transaction data, all we know is that it is not currently mined. It might still be in some other node's mempool.

Reply:
(This is the blocking comment)

Reply:
So this is correct?
```
is MainChain -> TransactionStatus.Mined(height.toBlockHeight(network))
is Mempool -> TransactionStatus.NotInMainChain
is OrphanedBlock -> TransactionStatus.NotInMainChain
```

Reply:
> So this is correct?
> 
> ```
> is MainChain -> TransactionStatus.Mined(height.toBlockHeight(network))
> is Mempool -> TransactionStatus.NotInMainChain
> is OrphanedBlock -> TransactionStatus.NotInMainChain
> ```

No, it isn't. All 3 cases of TransactionStatus represent a different state and those can't be merged. I don't know exactly what is marked by Kris as an issue. Maybe the terminology is weird. iOS doesn't use `MainChain`, `Mempool` and `OrphanedBlock`.

Reply:
All 3 cases do represent different states, but @nuttycom chose to explicitly merge `Mempool` and `OrphanedBlock` on the Rust side for now. This is partly due to a bug in zebrad (which backs some `lightwalletd` servers): it does not make this distinction (it currently reports `Mempool` _as_ `OrphanedBlock`, and treats the orphaned block state as `TxidNotRecognized`). To get around this bug state for now, we _do_ want to map both `Mempool` and `OrphanedBlock` to `TransactionStatus.NotInMainChain`; then later once the zebrad bug is fixed we can revisit this.

Reply:
It's also explicitly the case that an orphaned transaction _is_ recognized, i.e. the node has seen it, but it is not in the main chain. `TxidNotRecognized` is explicitly "the node does not know about this block." Where this distinction might become relevant is that if a the node doesn't know about a transaction, a client might (re)submit the transaction to the same node, whereas features around mempool rate limiting might prevent a recently evicted transaction from being reaccepted. In the future we might want an explicit `Mempool` state, but it'll take some more thought about how we want to treat orphaned transactions before the distinction between being in the mempool and being orphaned is particularly relevant. The main thing we might eventually want to do is perform extra checking to verify whether orphaned transactions belonging to the wallet have been invalidated by an anchor changing.

Reply:
```suggestion
        is OrphanedBlock -> TransactionStatus.NotInMainChain
```